src/kernel-install/60-ukify.install.in

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# -*- mode: python-mode -*-
+#
+# This file is part of systemd.
+#
+# systemd is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# systemd is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with systemd; If not, see <https://www.gnu.org/licenses/>.
+
+# pylint: disable=missing-docstring,invalid-name,import-outside-toplevel
+# pylint: disable=consider-using-with,unspecified-encoding,line-too-long
+# pylint: disable=too-many-locals,too-many-statements,too-many-return-statements
+# pylint: disable=too-many-branches,redefined-builtin,fixme
+
+import argparse
+import os
+import runpy
+import shlex
+import sys
+
+from pathlib import Path
+
+__version__ = '{{PROJECT_VERSION}} ({{GIT_VERSION}})'
+
+try:
+    VERBOSE = int(os.environ['KERNEL_INSTALL_VERBOSE']) > 0
+except (KeyError, ValueError):
+    VERBOSE = False
+
+# Override location of ukify and the boot stub for testing and debugging.
+UKIFY = os.getenv('KERNEL_INSTALL_UKIFY', '/usr/lib/systemd/ukify')
+BOOT_STUB = os.getenv('KERNEL_INSTALL_BOOT_STUB')
+
+
+def shell_join(cmd):
+    # TODO: drop in favour of shlex.join once shlex.join supports pathlib.Path.
+    return ' '.join(shlex.quote(str(x)) for x in cmd)
+
+def log(*args, **kwargs):
+    if VERBOSE:
+        print(*args, **kwargs)
```

Comment:
cpython's output is always buffered, maybe to be more shell like this should be
```suggestion
        print(*args, **kwargs, flush=True)
```

Reply:
Yeah, I'm not sure it's strictly necessary, I just remember running into this problem in the past, but that was code using multi processing, so it had a lot of children. 

An alternative to the explicit flush in the function, would be using cpython's `-u` option 
> Force the stdout and stderr streams to be unbuffered.  This option has no effect on the stdin stream.

This might be a bit hard, since using `/usr/bin/env` this would need the `-S` option, which should be now be available everywhere, though I'm not sure about.

On that note, I noticed that this plugin uses `env`. The other kernel-install plugins use `/bin/sh`, e.g. force the default system shell. Should this plugin maybe also default to the system python path?

Reply:
> Yeah, I'm not sure it's strictly necessary, I just remember running into this problem in the past, but that was code using multi processing, so it had a lot of children.
> 
> An alternative to the explicit flush in the function, would be using cpython's `-u` option
> 
> > Force the stdout and stderr streams to be unbuffered.  This option has no effect on the stdin stream.
> 
> This might be a bit hard, since using `/usr/bin/env` this would need the `-S` option, which should be now be available everywhere, though I'm not sure about.

Is there some problem that this is intended to solve? If not, I would keep default behaviour.
 
> On that note, I noticed that this plugin uses `env`. The other kernel-install plugins use `/bin/sh`, e.g. force the default system shell. Should this plugin maybe also default to the system python path?

We use `env` for python code. `/bin/bash` and `/bin/sh` are standarized locations, but the place where `python3` is installed varies. Some distros will replace the path in scripts to some desired system path. It is better leave it to them. For non-distro installations, using `env` is fine, it "just works".